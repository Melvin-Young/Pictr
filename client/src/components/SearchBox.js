import React from 'react';
import { delay } from '../utils/utils';

const SearchBox = (props) => {
	return (
		<div className="">
			<input
				className="searchTerm"
				type="search"
				placeholder="search: (e.g. red, fire truck)"
				onChange={props.handleSearch}
				onKeyUp={() => delay(function() {
					props.handleTimer();
				}, 1000)}
			/>
		</div>
	);
};

export default SearchBox;
