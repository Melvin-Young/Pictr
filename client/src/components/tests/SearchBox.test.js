import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import SearchBox from '../SearchBox';

Enzyme.configure({ adapter: new Adapter()});

describe('SearchBox', () => {
    it('should render', () => {
        const component = shallow(<SearchBox handleSearch={jest.fn()} handleTimer={jest.fn()}/>)
        expect(component).toMatchSnapshot();
    });
});
