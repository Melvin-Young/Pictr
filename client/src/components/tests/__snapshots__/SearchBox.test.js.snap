// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox
    handleSearch={[MockFunction]}
    handleTimer={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        className="searchTerm"
        onChange={[MockFunction]}
        onKeyUp={[Function]}
        placeholder="search: (e.g. red, fire truck)"
        type="search"
      />,
      "className": "",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "searchTerm",
        "onChange": [MockFunction],
        "onKeyUp": [Function],
        "placeholder": "search: (e.g. red, fire truck)",
        "type": "search",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="searchTerm"
          onChange={[MockFunction]}
          onKeyUp={[Function]}
          placeholder="search: (e.g. red, fire truck)"
          type="search"
        />,
        "className": "",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "searchTerm",
          "onChange": [MockFunction],
          "onKeyUp": [Function],
          "placeholder": "search: (e.g. red, fire truck)",
          "type": "search",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
